services/graph/pkg/unifiedrole/roles.go

```diff
@@ -531,39 +531,23 @@ func GetLegacyRoleName(role libregraph.UnifiedRoleDefinition) string {
 	return legacyNames[role.GetId()]
 }
 
-// weightRoles sorts the provided role definitions by the number of permissions[n].actions they grant,
-// the implementation is optimistic and assumes that the weight relies on the number of available actions.
+// weightRoles sorts the provided role definitions by the number of LibreGraphWeight,
 // descending - false - sorts the roles from least to most permissions
 // descending - true - sorts the roles from most to least permissions
 func weightRoles(roleSet []*libregraph.UnifiedRoleDefinition, constraints string, descending bool) []*libregraph.UnifiedRoleDefinition {
-	slices.SortFunc(roleSet, func(i, j *libregraph.UnifiedRoleDefinition) int {
-		var ia []string
-		for _, rp := range i.GetRolePermissions() {
-			if rp.GetCondition() == constraints {
-				ia = append(ia, rp.GetAllowedResourceActions()...)
-			}
-		}
+	slices.SortFunc(roleSet, func(a, b *libregraph.UnifiedRoleDefinition) int {
```

Comment:
I think you could replace the whole block below with:

```
if descending {
    return cmp.Compare(b.GetLibreGraphWeight(), b.GetLibreGraphWeight())
}
return cmp.Compare(a.GetLibreGraphWeight(), b.GetLibreGraphWeight())
```

Or even use `a.GetLibreGraphWeight() - b.GetLibreGraphWeight()` instead of calling `cmp.Compare`.